Bottom: bfac059c5cbcdc6e7a505abf3d04fee5493ca914
Top:    aeddeb44f5949fe8e78075954a58363a0d2ca020
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-12 20:02:41 +0800

Refresh of autofs-5.0.5-add-master-read-wait-option.patch

---

diff --git a/lib/defaults.c b/lib/defaults.c
index 7eabb70..0d1162c 100644
--- a/lib/defaults.c
+++ b/lib/defaults.c
@@ -295,181 +295,6 @@ struct list_head *defaults_get_uris(void)
 	return list;
 }
 
-<<<<<<< current
-=======
-/*
- * Read config env variables and check they have been set.
- *
- * This simple minded routine assumes the config file
- * is valid bourne shell script without spaces around "="
- * and that it has valid values.
- */
-unsigned int defaults_read_config(unsigned int to_syslog)
-{
-	FILE *f;
-	char buf[MAX_LINE_LEN];
-	char *res;
-
-	f = open_fopen_r(DEFAULTS_CONFIG_FILE);
-	if (!f)
-		return 0;
-
-	while ((res = fgets(buf, MAX_LINE_LEN, f))) {
-		char *key, *value;
-
-		if (!parse_line(res, &key, &value))
-			continue;
-
-		if (check_set_config_value(key, ENV_NAME_MASTER_MAP, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_TIMEOUT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_MASTER_WAIT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_NEGATIVE_TIMEOUT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_BROWSE_MODE, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_LOGGING, value, to_syslog) ||
-		    check_set_config_value(key, ENV_LDAP_TIMEOUT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_LDAP_NETWORK_TIMEOUT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_MAP_OBJ_CLASS, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_ENTRY_OBJ_CLASS, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_MAP_ATTR, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_ENTRY_ATTR, value, to_syslog) ||
-		    check_set_config_value(key, ENV_NAME_VALUE_ATTR, value, to_syslog) ||
-		    check_set_config_value(key, ENV_APPEND_OPTIONS, value, to_syslog) ||
-		    check_set_config_value(key, ENV_MOUNT_WAIT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_UMOUNT_WAIT, value, to_syslog) ||
-		    check_set_config_value(key, ENV_AUTH_CONF_FILE, value, to_syslog) ||
-		    check_set_config_value(key, ENV_MAP_HASH_TABLE_SIZE, value, to_syslog) ||
-		    check_set_config_value(key, ENV_MOUNT_NFS_DEFAULT_PROTOCOL, value, to_syslog))
-			;
-	}
-
-	if (!feof(f) || ferror(f)) {
-		if (!to_syslog) {
-			fprintf(stderr,
-				"fgets returned error %d while reading %s\n",
-				ferror(f), DEFAULTS_CONFIG_FILE);
-		} else {
-			logmsg("fgets returned error %d while reading %s",
-			      ferror(f), DEFAULTS_CONFIG_FILE);
-		}
-		fclose(f);
-		return 0;
-	}
-
-	fclose(f);
-	return 1;
-}
-
-const char *defaults_get_master_map(void)
-{
-	char *master;
-
-	master = get_env_string(ENV_NAME_MASTER_MAP);
-	if (!master)
-		return strdup(default_master_map_name);
-
-	return (const char *) master;
-}
-
-int defaults_master_set(void)
-{
-	char *val = getenv(ENV_NAME_MASTER_MAP);
-	if (!val)
-		return 0;
-
-	return 1;
-}
-
-unsigned int defaults_get_timeout(void)
-{
-	long timeout;
-
-	timeout = get_env_number(ENV_NAME_TIMEOUT);
-	if (timeout < 0)
-		timeout = DEFAULT_TIMEOUT;
-
-	return (unsigned int) timeout;
-}
-
-unsigned int defaults_get_master_wait(void)
-{
-	long wait;
-
-	wait = get_env_number(ENV_NAME_MASTER_WAIT);
-	if (wait < 0)
-		wait = DEFAULT_MASTER_WAIT;
-
-	return (unsigned int) wait;
-}
-
-unsigned int defaults_get_negative_timeout(void)
-{
-	long n_timeout;
-
-	n_timeout = get_env_number(ENV_NAME_NEGATIVE_TIMEOUT);
-	if (n_timeout <= 0)
-		n_timeout = DEFAULT_NEGATIVE_TIMEOUT;
-
-	return (unsigned int) n_timeout;
-}
-
-unsigned int defaults_get_browse_mode(void)
-{
-	int res;
-
-	res = get_env_yesno(ENV_NAME_BROWSE_MODE);
-	if (res < 0)
-		res = DEFAULT_BROWSE_MODE;
-
-	return res;
-}
-
-unsigned int defaults_get_logging(void)
-{
-	char *res;
-	unsigned int logging = DEFAULT_LOGGING;
-
-	res = get_env_string(ENV_NAME_LOGGING);
-	if (!res)
-		return logging;
-
-	if (!strcasecmp(res, "none"))
-		logging = DEFAULT_LOGGING;
-	else {
-		if (!strcasecmp(res, "verbose"))
-			logging |= LOGOPT_VERBOSE;
-
-		if (!strcasecmp(res, "debug"))
-			logging |= LOGOPT_DEBUG;
-	}
-
-	free(res);
-
-	return logging;
-}
-
-unsigned int defaults_get_ldap_timeout(void)
-{
-	int res;
-
-	res = get_env_number(ENV_LDAP_TIMEOUT);
-	if (res < 0)
-		res = DEFAULT_LDAP_TIMEOUT;
-
-	return res;
-}
-
-unsigned int defaults_get_ldap_network_timeout(void)
-{
-	int res;
-
-	res = get_env_number(ENV_LDAP_NETWORK_TIMEOUT);
-	if (res < 0)
-		res = DEFAULT_LDAP_NETWORK_TIMEOUT;
-
-	return res;
-}
-
->>>>>>> patched
 struct ldap_schema *defaults_get_default_schema(void)
 {
 	struct ldap_schema *schema;
@@ -692,6 +517,7 @@ unsigned int defaults_read_config(unsigned int to_syslog)
 
 		if (check_set_config_value(key, ENV_NAME_MASTER_MAP, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_TIMEOUT, value, to_syslog) ||
+		    check_set_config_value(key, ENV_NAME_MASTER_WAIT, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_NEGATIVE_TIMEOUT, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_BROWSE_MODE, value, to_syslog) ||
 		    check_set_config_value(key, ENV_NAME_LOGGING, value, to_syslog) ||
@@ -759,6 +585,17 @@ unsigned int defaults_get_timeout(void)
 	return (unsigned int) timeout;
 }
 
+unsigned int defaults_get_master_wait(void)
+{
+	long wait;
+
+	wait = get_env_number(ENV_NAME_MASTER_WAIT);
+	if (wait < 0)
+		wait = DEFAULT_MASTER_WAIT;
+
+	return (unsigned int) wait;
+}
+
 unsigned int defaults_get_negative_timeout(void)
 {
 	long n_timeout;
